File: Payhub - SDK/php_SDK/com/payhub/ws/model/Capture.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 27/07/2015
 * Time: 12:21
 */
class Capture extends WsConnections
{
    private $url_basePath;
    private $url;
    public $merchant;
    public $bill;
    public static $CAPTURE_ID_LINK = "capture/";
    public $transaction_id;

    /**
     * @return mixed
     */
    public function getUrlBasePath()
    {
        return $this->url_basePath;
    }

    /**
     * @param mixed $url_basePath
     */
    public function setUrlBasePath($url_basePath)
    {
        $this->url_basePath = $url_basePath;
    }

    /**
     * @return mixed
     */
    public function getUrl()
    {
        return $this->url_basePath."capture/";
    }

    /**
     * @param mixed $url
     */
    public function setUrl($url)
    {
        $this->url_basePath = $url;
    }

    /**
     * @return mixed
     */
    public function getMerchant()
    {
        return $this->merchant;
    }

    /**
     * @param mixed $merchant
     */
    public function setMerchant($merchant)
    {
        $this->merchant = $merchant;
    }

    /**
     * @return mixed
     */
    public function getBill()
    {
        return $this->bill;
    }

    /**
     * @param mixed $bill
     */
    public function setBill($bill)
    {
        $this->bill = $bill;
    }


    /**
     * Capture constructor.
     * @param $merchant
     * @param $bill
     * @param $transaction_id
     */
    public function __construct($merchant, $bill, $transaction_id)
    {
        if(!is_null($bill)){
            $this->bill = $bill;
        }
        if(!is_null($merchant)){
            $this->merchant = $merchant;
        }
        if(!is_null($transaction_id)){
            $this->transaction_id = $transaction_id;
        }
    }


    /**
     * @return mixed
     */
    public function getTransactionId()
    {
        return $this->transaction_id;
    }

    /**
     * @param mixed $transaction_id
     */
    public function setTransactionId($transaction_id)
    {
        $this->transaction_id = $transaction_id;
    }

    private function object_unset_nulls($obj)
    {
        if(is_null($obj)){$obj=$this;}
        $arrObj = is_object($obj) ? get_object_vars($obj) : $obj;
        foreach($arrObj as $key => $val)
        {
            $val = (is_array($val) || is_object($val)) ? $this->object_unset_nulls($val) : $val;
            if (is_array($obj))
                $obj[$key] = $val;
            else
                $obj->$key = $val;
            if($val == null)
                unset($obj->$key);
        }
        return $obj;
    }
    public function doCapture($request){
        $json = json_encode($this->object_unset_nulls(null));
        curl_setopt($request, CURLOPT_POSTFIELDS, $json);
        $result = $this->doPost($request,$this->getUrl());
        $response = CaptureResponseInformation::fromArray($result);
        return $response;
    }
}

File: Payhub - SDK/ruby_SDK/PayHubSDK/lib/PayHubSDK/com/payhub/ws/model/capture.rb
class Capture < WsConnections
  include JsonSerializer
  ATTRS=[:transaction_id, :bill, :merchant]
  attr_accessor :url_basePath,*ATTRS
  CAPTURE_ID_LINK = "capture/"
  @url=nil

  def initialize (merchant, transactionId, bill)
    Util::validate_params(self.class.name, merchant,bill)
    @merchant=merchant
    @transaction_id=transactionId
    @bill=bill
  end

  def url=(str)
    @url_basePath = str
  end

  def url
    return @url_basePath+CAPTURE_ID_LINK
  end

  def doCapture(http,request)
    json = self.serialize_to_json
    request.body = json
    response = JSON.parse(doPost(http,request))
    result=CaptureResponseInformation.new
    if not response.include?('errors')
      result=CaptureResponseInformation.from_json(JSON.generate(response))
    else
      errors ||= Array.new
      response['errors'].each do |error|
        errors_aux=Errors.from_json(JSON.generate(error))
        errors.push(errors_aux)
      end
      result.errors=errors
    end
    return result
  end

end